### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by jjames177**
# Submitting Week 1 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x ] Coding Concepts
- [x ] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Looks like we are missing all three files! Add, commit and push them up to this branch for review!

File: code-challenges.js [hunk 1]
```diff
@@ -17,33 +17,118 @@ const fruit4 = "kiwi"
...
+        return `${number} is at boiling point`  
+    } else {
+        return `${number} is below boiling point`
+    }
+}
+
+
+console.log(boilingPoint(temp1)) //Output = 42, is below boiling point
+console.log(boilingPoint(temp2)) //Output = 350, is above boiling point
+console.log(boilingPoint(temp3)) //Output = 212, is at boiling point
```
Comment (WalkerTexas-coder) [inline code-challenges.js:44]: Nicely done!

File: code-challenges.js [hunk 2]
```diff
@@ -17,33 +17,118 @@ const fruit4 = "kiwi"
...
+// to combine the newly reversed string into an array im going to use .join
+// i'm going to console.log(splitLetters.join('')) to see if it works
+
 const currentCohort = "Charlie 2022"
 
+const myCohort = "Charlie 2022"
+var splitLetters = myCohort.split('')
+splitLetters.reverse();
+
+console.log(splitLetters.join('')) // Output = 2202 eilrahC
```
Comment (WalkerTexas-coder) [inline code-challenges.js:80]: Good stuff. You are welcome to use variables that are given to you too.

File: code-challenges.js [hunk 3]
```diff
@@ -17,33 +17,118 @@ const fruit4 = "kiwi"
...
+        return number1 - number2
+    } else if(number1 === number2) {
+        return 0
+    } else {
+        return number2 - number1
+    }
+}
+console.log(smallerNumbers(number1, number2))  // Output = 42
+
+console.log(smallerNumbers(number3, number4)) // Output = 3
```
Comment (WalkerTexas-coder) [inline code-challenges.js:133]: Very cool

File: coding-concepts.js [hunk 4]
```diff
@@ -18,42 +19,53 @@ const colors = ["tangerine", "magenta", "lilac", "daffodil"]
...
+//     at Object.<anonymous> (/Users/learnacademy/Desktop/week-1-assessment-jjames177/coding-concepts.js:48:25)
+//     at Module._compile (node:internal/modules/cjs/loader:1105:14)
+//     at Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
+//     at Module.load (node:internal/modules/cjs/loader:981:32)
+//     at Module._load (node:internal/modules/cjs/loader:827:12)
+//     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
+//     at node:internal/main/run_main_module:17:47
+
+// Node.js v18.0.0
+
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:62]: Thanks for sharing the error code! Yea this one is tricksy on purpose. The goal here is that we recognize methods only work on certain data types that they are associated with.

File: interview-questions.md [hunk 5]
```diff
@@ -15,54 +15,54 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer: Remembering back from the notes i took, git is the technology aspect or the (tool). GitHub is a website that receives and stores the code or information. 
 
-  Researched answer:
+  Researched answer: Git is the tool that i mentioned in my answer that can have an unlimited amount of versions of code. GitHub is mainly used for sending repositories and updating them constantly, especially when i am pair programming.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:20]: Excellent! Yea, Git is a software that is installed on computers like any application.

File: interview-questions.md [hunk 6]
```diff
@@ -15,54 +15,54 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
...
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: I remember learning about iteration from our lecture that covered Javascript loops. Iteration is when a certain block of code is ran over and over until one thing is met which is the condition. 
 
-  Researched answer:
+  Researched answer: I learned that iterations are used in many computer languages and programs to perform important and specific task, complete problem solving, and when the iteration is complete it will give a solution to the user. A very common form of iteration is commonly referred to as a loop.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:44]: Nicely done!

**Review by WalkerTexas-coder — APPROVED** [review]
James you did a great job on this assessment! Keep up the good work into the next assessment. If you want, feel free to come up with multiple solutions to the code problems using different tool sets and document that process. Great work! Approved.

File: interview-questions.md (hunk 6 above)
Comment (jjames177) [reply interview-questions.md:44]: Thanks Austin!

**Pull request merged**